### ISSUE TITLE
Message null: command not found in syslog when missing fas_custom opion in opennds.conf 

### ISSUE THREAD

**Issue by doctor-ox** [openNDS/openNDS#322]
When fas_custom_variables_list  is missing in /etc/opennds/opennds.conf  or set to ''

    Mar  9 21:23:53 raspberrypi opennds[26574]: [5][Wed Mar 09 21:23:53 2022][26605](src/fw_iptables.c:1072) Authenticating 4.3.2.78 a6:62:09:c0:f0:f4
    Mar  9 21:23:53 raspberrypi opennds[26574]: /run/ndscids/TBmMGJhMTJhMmMyZWJmYjZkZDdiMjQzM2VjZWYyOGEzNzIwNDY1YzU0NmE0ODdiOTRmODdhMTNlZGYyNWUzNjY: line 16: null: command not found

Same with other fas_custom list.

Message is returned by /forward_authentication_service/binauth/binauth_log.sh, line 255  (ndsversion=9.6.1beta)

        . $mountpoint/ndscids/$cidfile

When option exist but with no value in /etc/opennds/opennds.conf, opennds does not start 
    
    Mar  9 21:20:51 raspberrypi opennds[26427]: [3][Wed Mar 09 21:20:51 2022][26427](src/conf.c:804) Option fas_custom_variables_list requires argument on line 59 in /etc/opennds/opennds.conf
    Mar  9 21:20:51 raspberrypi opennds[26427]: [3][Wed Mar 09 21:20:51 2022][26427](src/conf.c:805) Exiting...
    Mar  9 21:20:51 raspberrypi systemd[1]: opennds.service: Control process exited, code=exited, status=1/FAILURE
    Mar  9 21:20:51 raspberrypi systemd[1]: opennds.service: Failed with result 'exit-code'.
    Mar  9 21:20:51 raspberrypi systemd[1]: Failed to start openNDS Captive Portal.

**Comment by bluewavenet** [comment]
@doctor-ox 
Yes:
This is correctly reporting that a configured option with no value is an error.

**Comment by doctor-ox** [comment]
@bluewavenet 
Yes it is correctly reporting but it is wrongly creating a line "(null)" in cids file.
This line raises error message in the logs when executed with eval command (when authentication occurs)  
Not sure now that all the fas_customoptions  have the same behavior.  The fas_custom_files_list at least does. 

e.g. :   when fas_custom_files_list does not exist in /etc/opennds/opennds.conf file (I commented lines in conf)

 /etc/opennds/opennds.conf

    fas_custom_parameters_list input=voucher:Voucher:text;username:Nom:text
    fas_custom_parameters_list hidden=slength:Duration:hidden
    fas_custom_images_list Dragon_png=file:///etc/opennds/htdocs/images/Dragon.png
    fas_custom_images_list Dragon_jpg=file:///etc/opennds/htdocs/images/Dragon.jpg
    #fas_custom_variables_list fake=null
    #fas_custom_files_list fake=null

produces ncids file like 

    hid="1b13b7c8685da6a53e0e7f72de0ba78d434a2246df8f3ed5287c825ad348ac8b"
    clientip="4.3.2.78"
    clientmac="a6:62:01:c3:f3:e4"
    client_type="cpd_can"
    gatewayname="VSpot"
    gatewayurl="http%3a%2f%2fstatus.client"
    version="9.6.1beta"
    gatewayaddress="4.3.2.1:2050"
    gatewaymac="d0:37:45:da:b6:e0"
    originurl="http%3a%2f%2fconnectivitycheck.gstatic.com%2fgenerate_204"
    clientif="wlan1"
    themespec="/usr/lib/opennds/theme_vouchers.sh"
    hidden="slength:Duration:hidden"; input="voucher:Voucher:text;username:Nom:text";
    
    Dragon_jpg="file:///etc/opennds/htdocs/images/Dragon.jpg"; Dragon_png="file:///etc/opennds/htdocs/images/Dragon.png";
    (null)                                                  <=========== INVALID

**Comment by bluewavenet** [comment]
@doctor-ox 

> Yes it is correctly reporting but it is wrongly creating a line "(null)" in cids file.

If it is correctly reporting then a cids file will not be created at all as openNDS will abort its startup.

Is this a separate issue where a specified file does not exist?

BTW file:// is intended for use with a removable storage device eg usb stick to allow local updates by hot plugging some storage.

**Comment by bluewavenet** [comment]
@doctor-ox 
I guess that in libopennds.sh lines 176-179 it would be better if it checked if the file exists....

**Comment by doctor-ox** [comment]
@bluewavenet 

I haven't tested with a file that does not exist.

openNDS starts when option fas_custom_xxx_list is missing in my config file.  
openNDS does not start only when option line exists but is empty.

The (null) is writen because config.custom_files is not initialized. 

You have to ensure config.custom_xxx  are initialized before starting service
e.g. for custom_vars, to be repeated for all the custom lists:

    diff --git a/src/conf.c b/src/conf.c
    index d55710c..2805d55 100644
    --- a/src/conf.c
    +++ b/src/conf.c
    @@ -1200,6 +1200,11 @@ config_read(const char *filename)
                            break;
                    }
            }
    +        if ( config.custom_vars == NULL ) {
    +           debug(LOG_ERR, "config.custom_vars is NULL");
    +           debug(LOG_ERR, "Exiting...");
    +           exit(1);
    +        }
    
            fclose(fd);

You could also initialize them with  an empty string instead of a NULL.  That is simpler.

**Comment by bluewavenet** [comment]
@doctor-ox 

> openNDS starts when option fas_custom_xxx_list is missing in my config file.

As intended.

>  openNDS does not start only when option line exists but is empty.

Also as intended.

> The (null) is written because config.custom_files is not initialized.

Why is this a problem?
